restservice/src/main/java/org/itevents/util/OneTimePassword/OtpGenerator.java

```diff
@@ -0,0 +1,55 @@
+package org.itevents.util.OneTimePassword;
+
+import org.springframework.stereotype.Component;
+
+import java.util.Date;
+import java.util.UUID;
+
+@Component
+ public class OtpGenerator {
+
+    private String password;
+    private Date creationDate;
+    private Date expirationDate;
+
+    public String getPassword() {
+        return password;
+    }
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    public Date getCreationDate() {
+        return creationDate;
+    }
+
+    public void setCreationDate(Date creationDate) {
+        this.creationDate = creationDate;
+    }
+
+    public Date getExpirationDate() {
+        return expirationDate;
+    }
+
+    public void setExpirationDate(Date expirationDate) {
+        this.expirationDate = expirationDate;
+    }
+
+    public OtpGenerator generateOtp(long lifetimeInMinutes) {
+       return new OtpGenerator(lifetimeInMinutes);
+    }
+
+    public OtpGenerator() {
+    }
+
+    public OtpGenerator(long lifetimeInMinutes) {
+        setCreationDate(new Date());
+        if (lifetimeInMinutes>0) {
```

Comment:
Why do you use `generateOtp` method do get `otp`? You can use cunstructor instead of it and get otp via `new OtpGenerator(...)`. If you need different logic depends of `lifetimeInMinutes` use OOP: create overloaded constructors (`OtpGenerator()` and `OtpGenerator(lifetimeInMinutes)`) and implement your logic in them.

As I wrote before, `OtpGenerator` is inappropriate name for this class. Use instead `Otp` or `OneTimePassword`